### PR TITLE
Feat/transfer beneficiary

### PR DESCRIPTION
**Pull request by isaackps**

### CODE REVIEW

File: src/components/AssetManagementPanel/AssetManagementApplication/index.tsx [hunk 1]
```diff
@@ -37,12 +41,17 @@ export const AssetManagementApplication = ({
...
     setAssetManagementAction(AssetManagementActions);
     setHolderTransferringState("");
+    setBeneficiaryEndorseState("");
   };
 
   useEffect(() => {
     setHolderTransferringState(changeHolderState);
   }, [changeHolderState]);
 
+  useEffect(() => {
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementApplication/index.tsx]: why?????

File: src/components/AssetManagementPanel/AssetManagementApplication/index.tsx [hunk 2]
```diff
@@ -37,12 +41,17 @@ export const AssetManagementApplication = ({
   const onSetFormAction = (AssetManagementActions: AssetManagementActions) => {
     setAssetManagementAction(AssetManagementActions);
     setHolderTransferringState("");
+    setBeneficiaryEndorseState("");
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementApplication/index.tsx]: whyyyy?

File: src/components/AssetManagementPanel/AssetManagementApplication/index.tsx [hunk 3]
```diff
@@ -19,6 +19,7 @@ export const AssetManagementApplication = ({
 }: AssetManagementApplicationProps) => {
   const [assetManagementAction, setAssetManagementAction] = useState(AssetManagementActions.None);
   const [holderTransferringState, setHolderTransferringState] = useState("");
+  const [beneficiaryEndorseState, setBeneficiaryEndorseState] = useState("");
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementApplication/index.tsx]: whyyyy????

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx [hunk 4]
```diff
@@ -7,12 +7,12 @@ import { AssetManagementActions } from "./../../AssetManagementActions";
 interface ManageAssetsDropdownProps {
   onSetFormAction: (nextFormAction: AssetManagementActions) => void;
   className?: string;
-  canSurrender: boolean;
+  canEndorseAndSurrender: boolean;
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx]: why combine?

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx [hunk 5]
```diff
@@ -27,23 +27,22 @@ export const ManageAssetsDropdown = styled(
               Transfer holdership
             </Dropdown.Item>
           )}
-          {/*
-          {account === beneficiary && (
+
+          {canEndorseAndSurrender && (
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx]: why combine???

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx [hunk 6]
```diff
@@ -27,23 +27,22 @@ export const ManageAssetsDropdown = styled(
...
             <>
-              {!haveApprovedAddresses && (
-                <Dropdown.Item onClick={() => onSetFormAction(AssetManagementActions.EndorseBeneficiary)}>Endorse change of beneficiary</Dropdown.Item>
-              )}
+              {/* {!haveApprovedAddresses && ( */}
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx]: rm

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx [hunk 7]
```diff
@@ -27,23 +27,22 @@ export const ManageAssetsDropdown = styled(
...
+              <Dropdown.Item onClick={() => onSetFormAction(AssetManagementActions.EndorseBeneficiary)}>
+                Endorse change of beneficiary
+              </Dropdown.Item>
+              {/* )} */}
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx]: rm

File: src/components/AssetManagementPanel/AssetManagementForm/FormVariants/ActionSelectionForm/ActionSelectionForm.tsx [hunk 8]
```diff
@@ -29,11 +29,11 @@ export const ActionSelectionForm = ({
   beneficiary,
   holder,
   account,
-  canSurrender,
+  canEndorseAndSurrender,
   onConnectToWallet,
   canChangeHolder,
 }: ActionSelectionFormProps) => {
-  const canManage = canSurrender || canChangeHolder;
+  const canManage = canEndorseAndSurrender || canChangeHolder;
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementForm/FormVariants/ActionSelectionForm/ActionSelectionForm.tsx]: why combine?

File: src/components/AssetManagementPanel/AssetManagementForm/FormVariants/EndorseBeneficiary/EndorseBeneficiary.tsx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+          isSuccess: true,
+          beneficiaryAddress: newBeneficiary,
+        })
+      );
+    }
+  }, [isConfirmed, newBeneficiary, showOverlay]);
+
+  const isValidEndorse = () => {
+    if (!newBeneficiary || !newHolder) return false;
+    if (newBeneficiary === beneficiary && newHolder === holder) return false;
```
Comment (yehjxraymond) [inline src/components/AssetManagementPanel/AssetManagementForm/FormVariants/EndorseBeneficiary/EndorseBeneficiary.tsx:57]: might be weird if you fail silently here

Comment (yehjxraymond) [reply src/components/AssetManagementPanel/AssetManagementForm/FormVariants/EndorseBeneficiary/EndorseBeneficiary.tsx:57]: user who tried it wont know this condition

File: src/components/AssetManagementPanel/AssetManagementApplication/index.tsx (hunk 1 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementApplication/index.tsx]: ok i removed the extra state for the error state.

File: src/components/AssetManagementPanel/AssetManagementApplication/index.tsx (hunk 2 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementApplication/index.tsx]: same removed the extra state

File: src/components/AssetManagementPanel/AssetManagementApplication/index.tsx (hunk 3 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementApplication/index.tsx]: removed too

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx (hunk 4 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx]: ok u can decouple them but the checks are the same, we look for isBeneficiary and isHolder.

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx (hunk 5 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx]: ok decoupled it

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx (hunk 6 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx]: ok

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx (hunk 7 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementForm/AssetManagementDropdown/AssetManagementDropdown.tsx]: ok

File: src/components/AssetManagementPanel/AssetManagementForm/FormVariants/ActionSelectionForm/ActionSelectionForm.tsx (hunk 8 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementForm/FormVariants/ActionSelectionForm/ActionSelectionForm.tsx]: decoupled it

File: src/components/AssetManagementPanel/AssetManagementForm/FormVariants/EndorseBeneficiary/EndorseBeneficiary.tsx (hunk 9 above)
Comment (isaackps) [reply src/components/AssetManagementPanel/AssetManagementForm/FormVariants/EndorseBeneficiary/EndorseBeneficiary.tsx:57]: ok noted

**Review by yehjxraymond — APPROVED** [review]
LGTM

**Pull request merged**